packages/sync-engine/src/stripeSync.ts

```diff
@@ -1009,25 +1009,30 @@ export class StripeSync {
     fetch: () => Stripe.ApiListPromise<T>,
     upsert: (items: T[]) => Promise<T[]>
   ): Promise<Sync> {
-    const items: T[] = []
+    const chunkSize = 250
+    let chunk: T[] = []
+    let synced = 0
+    let pending: Promise<T[]> | null = null
```

Comment:
Instead of storing the pending promise here, couldnt we just await the upsert in line 1024 and then also remove line 1030? 🤔 

Could simplify this a little to avoid having multiple variables that need to be reset/stores in and after the loop

Overall approach lgtm though